### PR TITLE
write updating question endpoint

### PR DESCRIPTION
**Pull request by roza8765**

### CODE REVIEW

File: server/api/topics.js [hunk 1]
```diff
@@ -73,4 +73,19 @@ router.post("/:topic_id/question", (req, res) => {
 			res.json(500);
 		});
 });
+router.put("/:topic_id/question", function (req, res) {
+	const newQuestion = req.body.question_text;
+	const id = req.params.topic_id;
+	console.log(newQuestion);
```
Comment (myront) [inline server/api/topics.js]: it's better to remove the log statements, before merging. I know that it's useful for debugging,  bad is a bad practice leaving them in the code, as it's causing a lot of noise eventually

File: server/api/topics.js [hunk 2]
```diff
@@ -73,4 +73,19 @@ router.post("/:topic_id/question", (req, res) => {
...
+	if(!newQuestion){
+		return res.status(400).send("The question can not be empty");
+	}
+	topicDb
+		.updateQuestion( id, newQuestion)
+		.then(() => res.send("Question updated!"))
+	    .catch((err) => {
+			console.error(err);
```
Comment (myront) [inline server/api/topics.js]: Same here

File: server/api/topics.js (hunk 1 above)
Comment (teomrd) [inline server/api/topics.js]: it's better to remove the log statements, before merging. I know that it's useful for debugging, bad is a bad practice leaving them in the code, as it's causing a lot of noise eventually

File: server/api/topics.js (hunk 2 above)
Comment (teomrd) [inline server/api/topics.js]: Same here

File: server/api/topics.js [hunk 3]
```diff
@@ -73,4 +73,19 @@ router.post("/:topic_id/question", (req, res) => {
...
+router.put("/:topic_id/question", function (req, res) {
+	const newQuestion = req.body.question_text;
+	const id = req.params.topic_id;
+	console.log(newQuestion);
+	if(!newQuestion){
+		return res.status(400).send("The question can not be empty");
+	}
+	topicDb
+		.updateQuestion( id, newQuestion)
+		.then(() => res.send("Question updated!"))
```
Comment (teomrd) [inline server/api/topics.js]: As I have seen that you are sending responses in some different ways. `res.send`, `res.json`, `res.status(200).send`. Let's consolidate the response types

Always when sending returning responses contain a status code
* 200 OK (_on success_)
* 400 Bad request (_on  bad requests_)
* 401 Unauthorised (_on unauthenticated requests_)
... [More here](https://httpstatuses.com/)

And,
* Always response with a JSON response body.
In the case of create, update return the  created/updated entity

```js
res.status(200).json(updatedTopic);
```
In the error scenario,

```js
res.status(500).json({
  error: "Your error message that describes what went wrong"
});
```

File: server/api/topics.js [hunk 4]
```diff
@@ -59,6 +59,20 @@ router.get("/:topic_id/questions", (req, res) => {
...
+	const topicId = req.params.topic_id;
+	const questionId = req.params.question_id;
+	topicDb
+		.getQuestionById(topicId, questionId )
+		.then((data) => {
+			res.json(data);
+		})
+		.catch((err) => {
+			console.error(err);
+			res.json(500);
```
Comment (teomrd) [inline server/api/topics.js:88]: This is probably forgotten :)

File: server/api/topics.js [hunk 5]
```diff
@@ -67,10 +81,26 @@ router.post("/:topic_id/question", (req, res) => {
...
-		.then(() => res.send("Question created!"))
-		.catch((err) => {
-			console.error(err);
-			res.json(500);
-		});
+		.then((data) => res.status(200).json(data))
+		.catch(() => {
+			res.status(500).json({
+				error: "creating a new question failed",
+			});
```
Comment (teomrd) [inline server/api/topics.js:105]: Nice 👍

File: server/api/topics.js [hunk 6]
```diff
@@ -67,10 +81,26 @@ router.post("/:topic_id/question", (req, res) => {
...
+		 });
+});
+router.put("/:topic_id/question", function (req, res) {
+	const newQuestion = req.body.question_text;
+	const id = req.params.topic_id;
+	if(!newQuestion){
+		return res.status(400).send("The question can not be empty");
```
Comment (teomrd) [inline server/api/topics.js:112]: I suppose in here that can be consistent with the rest of the error responses 
```js
res.status(400).json({
  error: "The question can not be empty",
});
```

File: server/api/topics.js (hunk 4 above)
Comment (roza8765) [reply server/api/topics.js:88]: I am working in topics part now cause it is in another branch

File: server/api/topics.js (hunk 6 above)
Comment (roza8765) [reply server/api/topics.js:112]: I asked you about this and tell it is ok to write it like this

Comment (roza8765) [reply server/api/topics.js:112]: do u want me to change it

Comment (teomrd) [reply server/api/topics.js:112]: I think it's better. But if you are ok for with the way it is, I am merging

File: server/api/topics.js (hunk 4 above)
Comment (teomrd) [reply server/api/topics.js:88]: Make sure that you are working on a specific branch solving a specific problem. After being merged start working on a new card/branch. This way you can avoid having a git branch hell that you have half changes here half in there ;)

**Review by teomrd — DISMISSED** [review]

**Review by myront — APPROVED** [review]

**Pull request merged**